src/action/wallet.js

```diff
@@ -172,37 +185,81 @@ class WalletAction {
     await this.initWallet({
       walletPassword: password,
       seedMnemonic: this._store.seedMnemonic.toJSON(),
+      recoveryWindow: 0,
     });
   }
 
   /**
    * Initiate the lnd wallet using the generated seed and password. If this
    * is success set `walletUnlocked` to true and navigate to the seed success
-   * screen.
+   * screen. Otherwise, display an error and navigate back to the seed select
+   * view.
    * @param  {string} options.walletPassword The user chosen password
    * @param  {Array}  options.seedMnemonic   The seed words to generate the wallet
+   * @param  {number} options.recoveryWindow The number of addresses to recover
    * @return {Promise<undefined>}
    */
-  async initWallet({ walletPassword, seedMnemonic }) {
+  async initWallet({ walletPassword, seedMnemonic, recoveryWindow }) {
     try {
       await this._grpc.sendUnlockerCommand('InitWallet', {
         wallet_password: toBuffer(walletPassword),
         cipher_seed_mnemonic: seedMnemonic,
+        recovery_window: recoveryWindow,
       });
       this._store.walletUnlocked = true;
       this._nav.goSeedSuccess();
     } catch (err) {
-      this._notification.display({ msg: 'Initializing wallet failed', err });
+      this._nav.goSelectSeed();
+      this._notification.display({
+        type: 'error',
+        msg: 'Initializing wallet failed',
+      });
+    }
+  }
+
+  /**
+   * Initialize the restore wallet view by resetting input values and then
+   * navigating to the view.
+   * @return {undefined}
+   */
+  initRestoreWallet() {
+    this._store.wallet.seedVerify = Array(24).fill('');
+    this._store.wallet.restoreIndex = 0;
+    this._nav.goRestoreWallet();
+  }
+
+  /**
+   * Initialize the next restore wallet view by setting a new restoreIndex or,
+   * if all seed words have been entered, navigating to the password entry
+   * view.
+   * @return {undefined}
+   */
+  initNextRestorePage() {
+    if (this._store.wallet.restoreIndex < 21) {
+      this._store.wallet.restoreIndex += 3;
+    } else {
+      this._nav.goPassword();
     }
   }
 
   /**
    * Check the password input by the user by attempting to unlock the wallet.
+   * If we are currently restoring the wallet, initialize the wallet with
+   * the password input the seed that was already inputted, and the default
+   * recovery window.
    * @return {Promise<undefined>}
    */
   async checkPassword() {
     const { password } = this._store.wallet;
-    await this.unlockWallet({ walletPassword: password });
+    if (this._store.wallet.restoring) {
+      await this.initWallet({
+        walletPassword: password,
+        seedMnemonic: this._store.wallet.seedVerify.toJSON(),
+        recoveryWindow: RECOVERY_WINDOW,
+      });
+    } else {
+      await this.unlockWallet({ walletPassword: password });
+    }
```

Comment:
Not sure I understand. Why are we using the password check screen for restoration?